src/dom/dom.js

```diff
@@ -1009,11 +1009,11 @@ p5.prototype.createColorPicker = function(value) {
  * }
  * </code></div>
  */
-p5.prototype.createInput = function(value, type) {
+p5.prototype.createInput = function(value, type = 'text') {
   p5._validateParameters('createInput', arguments);
-  var elt = document.createElement('input');
-  elt.type = type ? type : 'text';
-  if (value) elt.value = value;
+  let elt = document.createElement('input');
+  elt.type = type;
+  if (value !== undefined) elt.value = value;
```

Comment:
I wonder if this check is necessary since `value` is a required parameter, it should never be `undefined`.

Reply:
Yes, it's not. I am updating it.

Reply:
What is the existing behaviour that would be broken by this change? ie. If `createInput()` is called without any arguments what does it do? If it doesn't do anything particularly useful I would say that's a bug and not a breaking change. Just in my opinion.

Reply:
But since, it's an input, which is literally used when we want the value from the user. So not all times, should it have a default value. I am having second thoughts about it. I think, i should revert it back to optional param and set default param as empty string. Even i am not sure.

Reply:
I only considered it as a breaking change because already declared `createInput` calls would start showing errors. So, i was wondering if there's some changelog, where i should mention that ? Or is it fine.

Reply:
In that case I think it should be reverted to being optional while implementing a check for if `value` is undefined it is set to an empty string. Would that still show errors if reverted like this?

Reply:
No, it won't. Even i feel like it should be reverted. And if reverted, it wouldn't show any errors. Cool, then it is resolved, value is optional.